src/components/atom/Modal/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { FC, ReactNode, useEffect } from "react";
+import { ReactComponent as ModalCloseIcon } from "../../../assets/img/modal-close-icon.svg";
+
+interface ModalProps {
+  isOpen: boolean;
+  modalHeaderText?: string;
+  onClose: () => void;
+  children: ReactNode;
+}
+
+const Modal: FC<ModalProps> = ({ isOpen, onClose, children, modalHeaderText }) => {
+  useEffect(() => {
```

Comment:
Can you create custom hook, syntax should be useOnPress("Escape", onClose) -> first parameter which key is pressed, second which function needs to be triggered.